core/trino-main/src/test/java/io/trino/block/BlockAssertions.java

```diff
@@ -312,7 +312,7 @@ else if (fieldValue instanceof Block) {
                     singleRowBlockWriter.appendStructure((Block) fieldValue);
                 }
                 else if (fieldValue instanceof Integer) {
-                    INTEGER.writeLong(singleRowBlockWriter, ((Integer) fieldValue).intValue());
+                    INTEGER.writeLong(singleRowBlockWriter, ((Integer) fieldValue).longValue());
```

Comment:
this looks like an unrelated change, please revert.

the code here can be simplified, let me do this in https://github.com/trinodb/trino/pull/8576